syntax/abstract.mjs

```diff
@@ -111,33 +111,38 @@ export function into(Type) {
     }
 }
 
+// Create a reducer suitable for joining adjacent nodes of the same type, if
+// type is one of types specified.
 function join_adjacent(...types) {
     return function(acc, cur) {
         let prev = acc[acc.length - 1];
         for (let Type of types) {
             if (prev instanceof Type && cur instanceof Type) {
-                join_of_type(Type, prev, cur);
+                // Replace prev with a new node of the same type whose value is
+                // the sum of prev and cur, and discard cur.
```

Comment:
... `discard cur and prev.`?

Reply:
Do you think it's needed given that the comment starts with `Replace...`?

Reply:
I could go without the `, and discard ...` altogether, too.